pandas/core/base.py

```diff
@@ -990,13 +1009,94 @@ def factorize(self, sort=False, na_sentinel=-1):
         from pandas.core.algorithms import factorize
         return factorize(self, sort=sort, na_sentinel=na_sentinel)
 
-    def searchsorted(self, key, side='left'):
-        """ np.ndarray searchsorted compat """
+    _shared_docs['searchsorted'] = (
+        """Find indices where elements should be inserted to maintain order.
+
+        Find the indices into a sorted %(klass)s `self` such that, if the
+        corresponding elements in `v` were inserted before the indices, the
+        order of `self` would be preserved.
+
+        Parameters
+        ----------
+        v : array_like
+            Values to insert into `a`.
+        side : {'left', 'right'}, optional
+            If 'left', the index of the first suitable location found is given.
+            If 'right', return the last such index.  If there is no suitable
+            index, return either 0 or N (where N is the length of `a`).
+        sorter : 1-D array_like, optional
```

Comment:
I either wouldn't even put this here at all, or put a comment that is for compatiblity with numpy; we are not advertising this at all. To be honest I really just like the strategy of not even having this in the signature, but allowing it to be passed (but must be ``None``). that way it doesn't need to be documented and is not confusing to the user.

We don't want to have keywords that are there just because numpy has them.